### PR TITLE
dev/core#511 Fix wrong current month showing on Membership Dashboard

### PR DESCRIPTION
**Pull request by aydun**
Overview
----------------------------------------
Correct the current month on the Membership Dashboard summary.

Before
----------------------------------------
screenshot from 2018-11-08 12-48-54

Current month is wrong

After
----------------------------------------
screenshot from 2018-11-08 12-48-22

Current month is right

Technical Details
----------------------------------------
Nothing much

Comments
----------------------------------------

https://lab.civicrm.org/dev/core/issues/511

### CODE REVIEW

File: CRM/Utils/Date.php [hunk 1]
```diff
@@ -438,6 +438,23 @@ public static function customFormat($dateString, $format = NULL, $dateParts = NU
...
+   * @param string $format
+   *   The output format.
+   * @param array $dateParts
+   *   An array with the desired date parts.
+   *
+   * @return string
+   *   the $format-formatted $date
+   */
+  public static function customFormatTs($timestamp, $format = NULL, $dateParts = NULL) {
+    return CRM_Utils_Date::customFormat(date("Y-m-d H:i:s", $timestamp)  , $format, $dateParts);
```
Comment (jitendrapurohit) [inline CRM/Utils/Date.php]: Maybe, the test is failing due to the extra space after `)`. Can you try removing that and ask jenkin to retest?

**Comment by jitendrapurohit** [comment]
Not sure if we are in need of an additional function here. Same could be handled in `Dashboard.php` by replacing the tpl variable assignment by -

    $this->assign('month', CRM_Utils_Date::customFormat(date('Y-m-d', $monthStartTs), '%B'));

But if this new function seems to be a good usage in future - Happy for this to merge as it fixes the main issue on Membership page as expected 👍

File: CRM/Utils/Date.php (hunk 1 above)
Comment (aydun) [reply CRM/Utils/Date.php]: Urgh!  Stray spaces removed.

**Comment by aydun** [comment]
Agreed, it could be handled directly in the assign but the function seemed like it might be useful elsewhere.

**Comment by eileenmcnaughton** [comment]
@aydun don't suppose we could get a test on that new function?

**Comment by aydun** [comment]
@eileenmcnaughton just for you :-)  With a bonus test for the existing customFormat() fn

**Comment by eileenmcnaughton** [comment]
@aydun so I requested a test well after you should have stopped working & you turned it around within an hour!!??!!??!!

Merge on pass I say!!

**Pull request merged**